### PR TITLE
feat(ui-shell): mobile nav enhacements

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes

- Adds social media icon links to mobile sidenav.
- Smallest breakpoint sidenav will take full screen width, tablet view will maintain max-width defined by carbon ui-shell
- Adds `Ecosystem` nav item to Community dropdown
- Fixes small classname typo `accout` -> `account`
- Includes small style update to ensure there the header content has a max width to be flush with page content on qiskit.org

## Implementation details
Followed similar styles to what was previously implemented in [navbar](https://github.com/Qiskit/web-components/tree/3671dd9d8c6f2da8eafeb093b905bf2f48565b43/components/navbar)

## Screenshots
Screen Shot 2022-09-09 at 1 50 13 PM

https://user-images.githubusercontent.com/8097306/189445937-5380f2c9-c977-47f0-8860-0c029ddb7704.mov

### CODE REVIEW

**Comment by yoshiemuranaka** [comment]
Seeing an error when running `npm run lint:eslint` locally–will investigate next week

File: components/ui-shell/index.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ import './header/index.js';
 import { qiskitLogoIcon } from '../icons/qiskit-logo.js';
 import { userIcon } from '../icons/user.js';
 import styles from './index.scss';
-import { NavItem, TopLevelNavItem, Variant, NAV_ITEMS } from './settings.js';
+import { Variant, NAV_ITEMS, SOCIAL_LINKS } from './settings.js';
+import type { NavItem, TopLevelNavItem } from './settings.js';
```
Comment (yoshiemuranaka) [inline components/ui-shell/index.ts:19]: Saw errors when trying to import type from the same statement 👇 

```ts
import { 
  Variant, 
  NAV_ITEMS, 
  SOCIAL_LINKS, 
  type NavItem,
  type TopLevelNavItem 
} from './settings.js';
```

File: components/ui-shell/index.scss [hunk 2]
```diff
@@ -378,3 +388,38 @@ $prefix: "bx";
     }
   }
 }
+
+.#{$q}-side-nav-footer {
+  @include carbon--breakpoint-up("md") {
```
Comment (techtolentino) [inline components/ui-shell/index.scss:393]: wondering if this should be "lg" and up?

reason: the mobile menu button is available at a certain breakpoint, and in that screen, I don't see the mobile menu footer.

https://user-images.githubusercontent.com/6276074/189955651-0b5cce3f-13c6-433e-afe9-e4db63c95bc2.mov

**Review by techtolentino — CHANGES_REQUESTED** [review]
Nice improvements here @yoshiemuranaka - thank you!

Just a small question regarding the mobile nav footer x breakpoints.

👏🏼

File: components/ui-shell/index.scss (hunk 2 above)
Comment (yoshiemuranaka) [reply components/ui-shell/index.scss:393]: I was wondering about this also, but stuck to the [original navbar implementation](https://github.com/Qiskit/web-components/blob/3671dd9d8c6f2da8eafeb093b905bf2f48565b43/components/navbar/index.scss#L70-L73)

And also noticed on `qiskit.org` we have the same behavior where smaller (tablet) breakpoint doesn't show the social footer, but the smallest (mobile) breakpoint does.

Screen Shot 2022-09-13 at 9 41 52 AM
Screen Shot 2022-09-13 at 9 42 00 AM


Do you think we should change this to show social links on the "tablet" view also?

Comment (techtolentino) [reply components/ui-shell/index.scss:404]: IMO, it's an improvement to the current implementation 

but I won't let it block the PR

Comment (techtolentino) [reply components/ui-shell/index.scss:404]: @JRussellHuffman maybe you have an opinion on this ☝🏼

Comment (techtolentino) [reply components/ui-shell/index.scss:404]: Resolving this conversation for now -- if we need to make updates, at least we're aware of where this is...

**Review by techtolentino — APPROVED** [review]
Good w/ all the changes here 

Great work!
🥇

File: components/ui-shell/index.ts (hunk 1 above)
Comment (abdonrd) [reply components/ui-shell/index.ts:19]: That's right... 🤔

Screen Shot 2022-09-14 at 11 12 45 AM

**Review by abdonrd — APPROVED** [review]
Good job, @yoshiemuranaka! 👏🏻

**Pull request merged**